gettsim/interface.py

```diff
@@ -65,55 +77,182 @@ def compute_taxes_and_transfers(
         DataFrame containing computed variables.
 
     """
+
+    # Set defaults for some parameters.
     targets = DEFAULT_TARGETS if targets is None else targets
     targets = parse_to_list_of_strings(targets, "targets")
     columns_overriding_functions = parse_to_list_of_strings(
         columns_overriding_functions, "columns_overriding_functions"
     )
     params = {} if params is None else params
 
+    # Load functions.
+    user_functions, internal_functions = load_user_and_internal_functions(functions)
+
+    # Perform several checks on functions and data. Merge internal and user functions.
+    all_functions = check_data_check_functions_and_merge_functions(
+        user_functions, internal_functions, columns_overriding_functions, data
+    )
+
+    # Set up dag.
+    dag = prepare_functions_and_set_up_dag(
+        all_functions=all_functions,
+        targets=targets,
+        params=params,
+        columns_overriding_functions=columns_overriding_functions,
+        check_minimal_specification=check_minimal_specification,
+        rounding=rounding,
+    )
+
+    # Do some checks.
+    _fail_if_root_nodes_are_missing(dag, data)
+    _fail_if_more_than_necessary_data_is_passed(dag, data, check_minimal_specification)
+
+    # We delay the data preparation as long as possible such that other checks can fail
```

Comment:
Comment does not seem quite up-to-date? We may want to do the `_process_data` step earlier with other checks? Looks a bit like it should be the first one, really.

E.g., `data_cols = list(data.columns)` in `check_data_check_functions_and_merge_functions` now implicitly assumes that the input data is a DataFrame.